### ISSUE TITLE
Double-sided material property not applied to objects loaded from gltf 

### ISSUE THREAD

**Issue by maria-korosteleva** [zauberzeug/nicegui#3515]
### Description

Hi all
I'm trying to display a garment 3D object, loaded as gltf. The material in my glb file is one-sided, so I tried to use the material settings from niceGUI to convert it to a double-sided material as follows:
```
# Camera setup
camera_location = [0, -200., 1.25]  
camera = ui.scene.perspective_camera(fov=np.pi / 6.)
camera.x = cam_location[0]
camera.y = cam_location[1]
camera.z = cam_location[2]
camera.look_at_x = 0
camera.look_at_y = 0
camera.look_at_z = cam_location[2] * 2/3

# Create scene
with ui.scene(
	width=1024,
	height=800, 
	camera=camera, 
	grid=False, 
	) as ui_3d_scene:
	ui_3d_scene.gltf(
		'geo/Configured_design_3D.glb', 
	).scale(0.01).rotate(np.pi / 2, 0., 0.).material(side='double')
```

However, the displayed object is still one-sided (the back of the garment visible through the neck and sleeve holes is not rendered): 
![image](https://github.com/user-attachments/assets/6dec9c9d-d696-4299-ac28-aa1d2702db6f) 

Here is how I expect it to look like: 
![image](https://github.com/user-attachments/assets/13f538f5-8b11-4be7-9bc2-eb4f35abccac)

Seems like currently, material setup in gltf object takes precedence over the additional material setup, but I would expect it to be the other way around, for the material properties that are explicitly set

Here is the glb file: [Uploading Configured_design_3D.zip…]()

**Comment by falkoschindler** [comment]
Thanks for reporting this issue, @maria-korosteleva!
Can you, please, check the glb upload? The link seems to be broken. Thanks!

**Comment by maria-korosteleva** [comment]
Hi @falkoschindler, oh, sorry, it should work now!

**Comment by fabian0702** [comment]
@maria-korosteleva The problem you're experiencing is most likely caused by nicegui loading gtlf's into a group that contains all of the gtlf's models. This would mean that in your case the code is trying to change the material of the group instead of the object itself (see [here](https://github.com/zauberzeug/nicegui/blob/main/nicegui/elements/scene.js#L271)).

**Comment by falkoschindler** [comment]
@fabian0702 seems to be right, the GLTF scene is wrapped in a `Group`:
https://github.com/zauberzeug/nicegui/blob/be557ffe43b9c4a177b5fa29a5f22182c045c999/nicegui/elements/scene.js#L269-L278
Do you have any idea how we could embed the file differently?

Apart from that: The `side` parameter can be "front", "back" or "both", but not "double". But it doesn't matter in this case.

**Comment by fabian0702** [comment]
@falkoschindler Given that gltf files can contain multiple objects, it can be difficult to change the material of just one object. Also, nicegui currently expects objects to be added to the scene immediately and the actual object data to be added later, which the current workaround solves.

**Comment by falkoschindler** [comment]
Ok, I see three problems:

1. Calling `material()` on a group (or GLTF object) does not work, because the JavaScript implementation of the `material()` method doesn't traverse children of groups.
2. Calling `material()` on a group (or GLTF object) does not work, because the group is added first and their children are added later. So traversing the group doesn't help, because there are no children yet.
3. Calling `material()` immediately after creating the GLTF object does not work, because the GLTF loader loads the object(s) asynchronously. Only when this is finished, `material()` can actually find the corresponding objects.

Problem 1 can be easily fixed.
Problem 2 is more challenging. We could instruct the user to add all children first before calling `material()` on a group. But this isn't very intuitive, as new children would be uncolored again.
Problem 3 is also tricky. But maybe the loader can transfer the material from the GLTF container onto the loaded objects.

Anyway, I wonder if there is a simpler solution somewhere in three.js. In PR #3549 for instance, we basically only had to change one flag to make groups draggable. Maybe there's something for letting child objects inherit their group's material?

**Comment by falkoschindler** [comment]
Here are three minimum examples for the above-mentioned problems:

```py
# Problem 1: Group material is not applied to child objects
with ui.scene() as scene:
    with scene.group() as group:
        scene.sphere()
    group.material('red')

# Problem 2: Group is still empty when applying material
with ui.scene() as scene:
    with scene.group().material('red'):
        scene.sphere()

# Problem 3: GLTF loader is not ready when applying material
avocado = 'https://raw.githubusercontent.com/KhronosGroup/glTF-Sample-Assets/main/Models/Avocado/glTF-Binary/Avocado.glb'
with ui.scene() as scene:
    scene.gltf(avocado).material('red').scale(100)
```

**Comment by falkoschindler** [comment]
Status update: PR #5708 will fix **case 3** (`.material()` called on a `gltf()`, both pre- and post-load). Cases 1 and 2 (material propagation from `group()` to its children) are being left unchanged by design, not deferred: Three.js itself doesn't propagate material from a parent group to its children, so any group-inheritance model would be something we'd invent and maintain on top. Without a concrete driver we'd rather stay consistent with Three.js semantics — apply material to mesh objects directly.

This issue will be closed once #5708 lands.
